lib/customer_repository.rb

```diff
@@ -0,0 +1,76 @@
+require_relative 'customer'
+
+class CustomerRepository
+  attr_reader :customers
+
+  def initialize(file_path, engine)
+    @engine = engine
+    @customers = create_repository(file_path)
+  end
+
+  def create_repository(file_path)
+    file = CSV.readlines(file_path, headers: true, header_converters: :symbol)
+    file.map do |row|
+      Customer.new(row)
+    end
+  end
+
+  def inspect
+    "#<#{self.class} #{@items.size} rows>"
+  end
+
+  def all
+    @customers
+  end
+
+  def find_by_id(id)
+    @customers.find do |customer|
+      customer.id.to_i == id
+    end
+  end
+
+  def find_all_by_first_name(name)
+      @customers.find_all do |customer|
+        customer.first_name.casecmp(name) == 0
+      end
+    end
+
+  def find_all_by_last_name(name)
+      @customers.find_all do |customer|
+        customer.last_name.casecmp(name) == 0
+      end
+    end
+
+    def max_customer_id
+      @customers.max_by do |customer|
+        customer.id
+      end.id
+    end
+
+  def create(attributes)
```

Comment:
- [ ] This one will need to be changed to match the rest of the features and what lovely Hound is saying.

Reply:
For this we need to make sure that our first name and last name are using the attributes method.
So inside the hash your pushing into the customer repository, it would be first_name: attributes[:first_name] and last_name: attributes[:last_name].